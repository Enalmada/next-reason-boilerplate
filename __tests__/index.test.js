/* eslint-env jest */

import {shallow} from "enzyme";
import React from "react";
import {MockedProvider} from "react-apollo/test-utils";
import createComponentWithPrerequisites from "../util/test/createComponentWithPrerequisites";
import App from "../pages/index.bs";

// The component AND the query need to be exported
const mocks = [];


describe("With Enzyme", () => {
    it("App shows \"Hello world!\"", () => {
        const app = shallow(<App/>).dive();

        expect(app.find("h1").text()).toEqual("nextjs reason-react boilerplate demonstrating some concepts");
    });
});

describe("With Snapshot Testing", () => {
    it("App shows \"Hello world!\"", () => {
        const component = createComponentWithPrerequisites(<MockedProvider mocks={mocks} addTypename={false}>
            <App/></MockedProvider>);
        expect(component.text()).toMatchSnapshot();
    });
});
