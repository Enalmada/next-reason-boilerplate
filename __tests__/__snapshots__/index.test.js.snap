// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello world!" 1`] = `"···Index···Reducer···Sentry···Ant Desktop···Antd Mobile···Health···intl···Sign in···nextjs reason-react boilerplate demonstrating some conceptsFont awesome icon loaded without flickerrendering based on user agent (use chrome devtools pass iphone, etc)This will be rendered on desktop!!react-intl translationsHello, Default!Link with title using intl api for stringAuthHello Preload getInitialProps on hover (* only in production)hover megetInitialPropsonServer: falseContext1000Sticky Footer Example © 2018"`;
